File: rest-common/src/main/java/org/chronopolis/rest/models/storage/StorageRegion.java
package org.chronopolis.rest.models.storage;


/**
 * Model for our StorageRegion entity
 *
 * Created by shake on 7/11/17.
 */
public class StorageRegion {

    private Long id;
    private String node;
    private String note;
    private Long capacity;
    private DataType dataType;
    private StorageType storageType;
    private ReplicationConfig replicationConfig;

    public String getNode() {
        return node;
    }

    public StorageRegion setNode(String node) {
        this.node = node;
        return this;
    }

    public Long getCapacity() {
        return capacity;
    }

    public StorageRegion setCapacity(Long capacity) {
        this.capacity = capacity;
        return this;
    }

    public StorageType getStorageType() {
        return storageType;
    }

    public StorageRegion setStorageType(StorageType storageType) {
        this.storageType = storageType;
        return this;
    }

    public ReplicationConfig getReplicationConfig() {
        return replicationConfig;
    }

    public StorageRegion setReplicationConfig(ReplicationConfig replicationConfig) {
        this.replicationConfig = replicationConfig;
        return this;
    }

    public Long getId() {
        return id;
    }

    public StorageRegion setId(Long id) {
        this.id = id;
        return this;
    }

    public DataType getDataType() {
        return dataType;
    }

    public StorageRegion setDataType(DataType dataType) {
        this.dataType = dataType;
        return this;
    }

    public String getNote() {
        return note;
    }

    public StorageRegion setNote(String note) {
        this.note = note;
        return this;
    }
}

File: rest-entities/src/main/kotlin/org/chronopolis/rest/kot/entities/storage/StorageRegion.kt
package org.chronopolis.rest.kot.entities.storage

import org.chronopolis.rest.kot.entities.Node
import org.chronopolis.rest.kot.entities.UpdatableEntity
import org.chronopolis.rest.kot.models.enums.DataType
import org.chronopolis.rest.kot.models.enums.StorageType
import javax.persistence.CascadeType
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.FetchType
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.persistence.OneToOne

@Entity
class StorageRegion(
        @Enumerated(value = EnumType.STRING)
        var dataType: DataType = DataType.BAG,

        @Enumerated(value = EnumType.STRING)
        var storageType: StorageType = StorageType.LOCAL,

        @OneToMany(mappedBy = "region", fetch = FetchType.LAZY)
        var storage: MutableSet<StagingStorage> = mutableSetOf(),

        var capacity: Long = 0,
        var note: String = ""
) : UpdatableEntity() {
        @ManyToOne
        lateinit var node: Node

        @OneToOne(mappedBy = "region", cascade = [CascadeType.ALL])
        lateinit var replicationConfig: ReplicationConfig
}